pkg/webservice/handlers.go

```diff
@@ -236,3 +236,39 @@ func getNodeJSON(nodeInfo *cache.NodeInfo) *dao.NodeDAOInfo {
 		Schedulable: nodeInfo.IsSchedulable(),
 	}
 }
+
+func GetApplicationHistory(w http.ResponseWriter, r *http.Request) {
+	writeHeaders(w)
+
+	var result []*dao.ApplicationHistoryDAOInfo
+	records := imHistory.GetRecords()
+	for _, record := range records {
```

Comment:
Need to check the impact of the earlier mentioned unlimited growth solution on this iteration loop

Reply:
Could you check whether the fixing commit is good? I hope I get your first item right.